vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -69,12 +70,22 @@
   private Set<String> compressedMediaTypes = Collections.emptySet();
   private Set<String> compressedFileSuffixes = Collections.emptySet();
 
+  private final ClassLoader classLoader;
   private final FSTune tune = new FSTune();
   private final FSPropsCache cache = new FSPropsCache();
+  private FileSystem fileSystem;
 
-  private String directoryTemplate(Vertx vertx) {
+  public StaticHandlerImpl() {
+    this.classLoader = null;
```

Comment:
Maybe capture the TCCL here. 
It would ease OSGi integration where the router is generally called with the Bundle classloader as TCCL